homeassistant/components/google_assistant/http.py

```diff
@@ -219,6 +236,56 @@ async def async_report_state(self, message, agent_user_id: str):
         }
         await self.async_call_homegraph_api(REPORT_STATE_BASE_URL, data)
 
+    @callback
+    def async_enable_local_sdk(self):
+        """Enable the local SDK."""
+        for key, value in self._store.agent_user_ids.items():
+            try:
+                webhook.async_register(
+                    self.hass,
+                    DOMAIN,
+                    "Local Support for " + key,
+                    value[STORE_GOOGLE_LOCAL_WEBHOOK_ID],
+                    self._handle_local_webhook,
+                )
+            except ValueError:
+                _LOGGER.info("Webhook handler is already defined!")
+                return
+
+        self._local_sdk_active = True
+
+    @callback
+    def async_disable_local_sdk(self):
+        """Disable the local SDK."""
+        for _, value in self._store.agent_user_ids.items():
+            webhook.async_unregister(self.hass, value[STORE_GOOGLE_LOCAL_WEBHOOK_ID])
+        self._local_sdk_active = False
+
+    async def _handle_local_webhook(self, hass, webhook_id, request):
+        """Handle an incoming local SDK message."""
+        # Circular dep
+        # pylint: disable=import-outside-toplevel
+        from . import smart_home
+
+        payload = await request.json()
+
+        if _LOGGER.isEnabledFor(logging.DEBUG):
+            _LOGGER.debug("Received local message:\n%s\n", pprint.pformat(payload))
+
+        for key, value in self._store.agent_user_ids.items():
```

Comment:
Make sure you always define the variable or it might not be defined:
```suggestion
				agent_user_id = None
        for key, value in self._store.agent_user_ids.items():
```

Reply:
moved to `get_local_agent_user_id` but implemented this.